src/index.ts

```diff
@@ -160,76 +238,96 @@ export const getNPMMetadataForDep = async (dep, npmAuthToken?: string) => {
+  { renderMessage = defaultRenderMessage }: Options = {},
+): void => {
   const sentence = danger.utils.sentence
 
   if (packageDiff.dependencies && packageDiff.dependencies.added) {
-    const typesDeps = packageDiff.dependencies.added.filter(d => d.startsWith("@types")).map(printDep)
+    const typesDeps = packageDiff.dependencies.added.filter(d => d.startsWith("@types")).map(escapePackage)
     if (typesDeps.length) {
-      const message = `@types dependencies were added to package.json, as a dependency for others.`
+      const msg = `@types dependencies were added to package.json, as a dependency for others.`
       const idea = `You need to move ${sentence(typesDeps)} into "devDependencies"?`
-      fail(`${message}<br/><i>${idea}</i>`)
+      fail(renderMessage(msg, idea))
     }
   }
 }
 
 export interface Options {
-  pathToPackageJSON?: string
+  pathToRootPackageJSON?: string
+  pathToPackageJSON?: string | ReadonlyArray<string>
   npmAuthToken?: string
+  renderPackageDetails?: typeof defaultRenderPackageDetails
+  renderReadme?: typeof defaultRenderReadme
+  renderYarnWhy?: typeof defaultRenderYarnWhy
+  renderMessage?: typeof defaultRenderMessage
+  renderPackageLink?: typeof defaultRenderPackageLink
 }
 
 /**
  * Provides dependency information on dependency changes in a PR
  */
-export default async function yarn(options: Options = {}) {
-  const path = options.pathToPackageJSON ? options.pathToPackageJSON : "package.json"
-  const packageDiff = await danger.git.JSONDiffForFile(path)
+export default async function yarn(options: Options = {}): Promise<void> {
+  const pathToRootPackageJSON = options.pathToRootPackageJSON ? options.pathToRootPackageJSON : "package.json"
+
+  let pathToPackageJSONs: ReadonlyArray<string>
+  if (options.pathToPackageJSON) {
+    pathToPackageJSONs = Array.isArray(options.pathToPackageJSON)
+      ? options.pathToPackageJSON
+      : [options.pathToPackageJSON]
+  } else {
+    const isNonRootPackageJson = (path: string) => path.endsWith("package.json") && path !== pathToRootPackageJSON
+
+    pathToPackageJSONs = Array.from(
+      new Set([...danger.git.created_files, ...danger.git.modified_files].filter(isNonRootPackageJson)),
+    )
+  }
+
+  const rootPackageDiff = await danger.git.JSONDiffForFile(pathToRootPackageJSON)
+  const packageDiffs: ReadonlyArray<JSONDiff> = await Promise.all(
+    pathToPackageJSONs.map(
+      (pathToPackageJSON: string): Promise<JSONDiff> => danger.git.JSONDiffForFile(pathToPackageJSON),
+    ),
+  )
 
-  checkForRelease(packageDiff)
-  checkForLockfileDiff(packageDiff)
-  checkForTypesInDeps(packageDiff)
+  checkForLockfileDiff(rootPackageDiff)
 
-  await checkForNewDependencies(packageDiff, options.npmAuthToken)
+  const config = await getYarnConfig()
+  for (const packageDiff of packageDiffs) {
+    checkForRelease(packageDiff, options)
+    checkForTypesInDeps(packageDiff, options)
+
+    await checkForNewDependencies(packageDiff, config, options)
```

Comment:
I think that when there are multiple packages which add dependencies this should make sure they each get handled, however I need to check that it works as expected...

Reply:
👍